sdks/java/io/google-ads/src/main/java/org/apache/beam/sdk/io/googleads/GoogleAdsV14.java

```diff
@@ -0,0 +1,647 @@
+     * {@link GoogleAdsClientFactory} builds the {@link GoogleAdsClient} used to construct service
+     * clients. The {@link DefaultGoogleAdsClientFactory} should be sufficient for most purposes
+     * unless the construction of {@link GoogleAdsClient} requires customization.
+     *
+     * @param googleAdsClientFactory
+     * @return A new {@link ReadAll} transform with the specified client factory.
+     * @see GoogleAdsClient
+     */
+    public ReadAll withGoogleAdsClientFactory(GoogleAdsClientFactory googleAdsClientFactory) {
+      checkArgumentNotNull(googleAdsClientFactory, "googleAdsClientFactory cannot be null");
+
+      return toBuilder().setGoogleAdsClientFactory(googleAdsClientFactory).build();
+    }
+
+    /**
+     * Creates and returns a new {@link ReadAll} transform with the specified rate limit policy
+     * factory. A {@link RateLimitPolicyFactory} builds the {@link RateLimitPolicy} used to limit
+     * the number of requests made by {@link ReadAll.ReadAllFn}. The Google Ads API enforces global
+     * limits from the developer token down to the customer ID and it is recommended to host a
+     * shared rate limiting service to coordinate traffic to the Google Ads API across all
+     * applications using the same developer token. Users of these transforms are strongly advised
+     * to implement their own {@link RateLimitPolicy} and {@link RateLimitPolicyFactory} to interact
+     * with a shared rate limiting service for any production workloads.
+     *
+     * @param rateLimitPolicyFactory
+     * @return A new {@link ReadAll} transform with the specified rate limit policy factory.
+     * @see GoogleAdsClient
+     */
+    public ReadAll withRateLimitPolicy(RateLimitPolicyFactory rateLimitPolicyFactory) {
+      checkArgumentNotNull(rateLimitPolicyFactory, "rateLimitPolicyFactory cannot be null");
+
+      return toBuilder().setRateLimitPolicyFactory(rateLimitPolicyFactory).build();
+    }
+
+    @Override
+    public PCollection<GoogleAdsRow> expand(PCollection<SearchGoogleAdsStreamRequest> input) {
+      GoogleAdsOptions options = input.getPipeline().getOptions().as(GoogleAdsOptions.class);
+
+      checkArgument(
+          options.getGoogleAdsDeveloperToken() != null || getDeveloperToken() != null,
+          "either --googleAdsDeveloperToken or .withDeveloperToken() is required");
+
+      return input.apply(ParDo.of(new ReadAllFn(this)));
+    }
+
+    /**
+     * A {@link DoFn} that reads reports from Google Ads for each query using the {@code
+     * SearchStream} method.
+     */
+    @VisibleForTesting
+    static class ReadAllFn extends DoFn<SearchGoogleAdsStreamRequest, GoogleAdsRow> {
+      private static final int MAX_RETRIES = 5;
+      private static final FluentBackoff BACKOFF =
+          FluentBackoff.DEFAULT
+              .withExponent(2.0)
+              .withInitialBackoff(Duration.standardSeconds(30))
+              .withMaxRetries(MAX_RETRIES);
```

Comment:
nit Are those arbitrary or recommended somewhere? If recommended, probably a reference would make sense

Reply:
The retry configuration is an adaption of the BQ DTS configuration. I've added a comment to reflect that these aren't entirely arbitrary.